user_tools/src/spark_rapids_pytools/cloud_api/dataproc.py

```diff
@@ -158,9 +158,30 @@ def get_matching_executor_instance(self, cores_per_executor):
         executors_from_config = self.configs.get_value('clusterInference', 'defaultCpuInstances', 'executor')
         # TODO: Currently only single series is supported. Change this to a loop when using multiple series.
         series_name, unit_info = list(executors_from_config.items())[0]
+        # If the number of cores is a valid entry in the list, return the equivalent instance
         if cores_per_executor in unit_info['vCPUs']:
             return f'{series_name}-{cores_per_executor}'
-        return None
+        # If the num-cores is not in the list, then we need to adjust the cores to the best match
+        # that is greater than the input cores.
+        rev_sorted_cores = unit_info['vCPUs'].copy()
+        rev_sorted_cores.sort(reverse=True)
+        adjusted_cores = rev_sorted_cores[0]
+        for num_cpu in rev_sorted_cores[1:]:
+            if num_cpu >= cores_per_executor:
+                adjusted_cores = num_cpu
+        # At this point adjusted_cores should be the best match.
+        selected_machine_type = f'{series_name}-{adjusted_cores}'
+        self.logger.info('The number of cores %d is not in the list of supported cores.',
+                         cores_per_executor)
+        if adjusted_cores > cores_per_executor:
```

Comment:
I find the log message slightly confusing. Can we have something as:

`Adjusting number of cores to match the nearest CSP machine with higher number of cores`

OR

`Adjusting machine type to match the nearest CSP machine with higher number of cores`


Similarly for the else part.